src/test/rustdoc-ui/intra-links-ambiguity.stderr

```diff
@@ -18,21 +18,6 @@ help: to link to the function, add parentheses
 LL | /// [`ambiguous()`] is ambiguous.
    |      ^^^^^^^^^^^^^
 
-error: `ambiguous` is both a struct and a function
-  --> $DIR/intra-links-ambiguity.rs:29:6
-   |
-LL | /// [ambiguous] is ambiguous.
-   |      ^^^^^^^^^ ambiguous link
-   |
-help: to link to the struct, prefix with `struct@`
-   |
-LL | /// [struct@ambiguous] is ambiguous.
-   |      ^^^^^^^^^^^^^^^^
-help: to link to the function, add parentheses
-   |
-LL | /// [ambiguous()] is ambiguous.
-   |      ^^^^^^^^^^^
-
 error: `multi_conflict` is a struct, a function, and a macro
```

Comment:
@jyn514 If I use `path_str` in the key, these errors are not reported, i.e. ambiguity errors aren't reported twice.

These errors were only emitted before (with orig_link), because one of the texts was surrounded with backticks, and the other wasn't.

Reply:
Ok I think I see the problem: there are two things we're trying to do and I kind of combined them in the issue.

1. Cache links and resolutions. This should be based on the things I said above, `path_str`, etc.
2. Don't report duplicate errors _for reference links_. This means we need to distinguish between reference links and regular links.

I think what we should do is turn this PR to _only_ fixing 1, and still report resolution failures even if the failure is cached. Then we can fix 2. in a follow-up that handles different types of links properly.

Reply:
Ah I should've realized that. That's 1:0 for tunnel vision.